### PR TITLE
add subject disposition

### LINKED ISSUE #2003: Feature: Add subject disposition.
## Feature Details
Participant status can be one of:
- Active
- Completed
- Discontinued
- Screen Failure
- In Screening

### PR DESCRIPTION
**Pull request by samussiah**
## Overview
This branch adds subject-level status using the study completion domain, then aggregates at the study, site, and country levels.

## Test Notes/Sample Code
```
lSourceData <- list(
    Raw_STUDY = clindata::ctms_study,
    Raw_SITE = clindata::ctms_site,
    Raw_SUBJ = clindata::rawplus_dm,
    Raw_STUDCOMP = clindata::rawplus_studcomp
)
lWorkflows <- MakeWorkflowList(c('STUDY', 'SITE', 'COUNTRY', 'SUBJ', 'STUDCOMP'))
lSpec <- CombineSpecs(lWorkflows)
lRawData <- Ingest(lSourceData, lSpec)
lMappedData <- RunWorkflows(
    lWorkflows,
    lRawData
)
dfGroups <- RunWorkflow(
    MakeWorkflowList('Groups')[[1]],
    lMappedData
)
dfGroups %>% filter(Param == 'ActiveParticipants')
```

## Connected Issues

- Closes #2003

### CODE REVIEW

File: inst/workflow/1_mappings/SITE.yaml [hunk 1]
```diff
@@ -35,6 +36,8 @@ spec:
       type: character
     subjid:
       type: character
+    Status:
```
Comment (lauramaxwell) [inline inst/workflow/1_mappings/SITE.yaml:39]: we can remove this Status, right? looks like a duplicate

File: inst/workflow/1_mappings/STUDY.yaml [hunk 2]
```diff
@@ -41,6 +42,8 @@ spec:
       type: character
     subjid:
       type: character
+    Status:
```
Comment (lauramaxwell) [inline inst/workflow/1_mappings/STUDY.yaml:45]: Another duplicate Status as far as i can tell

**Review by lauramaxwell — CHANGES_REQUESTED** [review]
A few tiny yaml edits that will also require a rerun of tests to update snaps, but other than that (and some investigation into the check failueres), this is looking good!

File: inst/workflow/1_mappings/STUDY.yaml (hunk 2 above)
Comment (samussiah) [reply inst/workflow/1_mappings/STUDY.yaml:45]: Nice catch!

Comment (samussiah) [reply inst/workflow/1_mappings/STUDY.yaml:45]: Oh wait, `Mapped_SUBJ$Status` is used to calculate the number of active subjects.  `Raw_STUDY$Status` is a study-level attribute.

File: inst/workflow/1_mappings/SITE.yaml (hunk 1 above)
Comment (samussiah) [reply inst/workflow/1_mappings/SITE.yaml:39]: This one captures subject status, to count the number of active subjects at the site.

**Comment by samussiah** [comment]
This PR kind of standardizes `Status` so the study and each site/participant will have a status, because the RBQM team can only follow up with active sites and participants.

**Pull request merged**